cmd/puppeth/genesis.go

```diff
@@ -174,7 +174,11 @@ func (spec *alethGenesisSpec) setPrecompile(address byte, data *alethGenesisSpec
 	if spec.Accounts == nil {
 		spec.Accounts = make(map[common.UnprefixedAddress]*alethGenesisSpecAccount)
 	}
-	spec.Accounts[common.UnprefixedAddress(common.BytesToAddress([]byte{address}))].Precompiled = data
+	a := common.UnprefixedAddress(common.BytesToAddress([]byte{address}))
+	if _, exist := spec.Accounts[a]; !exist {
+		spec.Accounts[a] = &alethGenesisSpecAccount{}
+	}
+	spec.Accounts[a].Precompiled = data
```

Comment:
I think we can make this simpler (there's no real reason to handle duplicate sets meaningfully):

```go
addr := common.UnprefixedAddress(common.BytesToAddress([]byte{address}))
spec.Accounts[addr] = &alethGenesisSpecAccount{
	Precompiled: data,
}
```

Reply:
okay, I updated as per your request

Reply:
updated code make the test failed then I reverted to the old code and it's okay because the parity genesis also do like that

```
	a := common.UnprefixedAddress(common.BytesToAddress([]byte{address}))
	if _, exist := spec.Accounts[a]; !exist {
		spec.Accounts[a] = &parityChainSpecAccount{}
	}
	spec.Accounts[a].Builtin = data
```

Reply:
Well yeah, but you also changes `Precompiled` to `Builtin`. If you use `Builtin` in my suggested code, it should be the same.